libs/core/include/core/fetch_backward.hpp

```diff
@@ -0,0 +1,223 @@
+      SIGEMT,  // emulation instruction executed
+#endif
+    };
+    return std::vector<int>(posix_signals,
+                            posix_signals + sizeof posix_signals / sizeof posix_signals[0]);
+  }
+
+  FetchBackward(const std::vector<int> &posix_signals = make_default_signals())
+    : _loaded(false)
+  {
+    bool success = true;
+
+    const size_t stack_size = 1024 * 1024 * 8;
+    _stack_content.reset(static_cast<char *>(malloc(stack_size)));
+    if (_stack_content)
+    {
+      stack_t ss;
+      ss.ss_sp    = _stack_content.get();
+      ss.ss_size  = stack_size;
+      ss.ss_flags = 0;
+      if (sigaltstack(&ss, nullptr) < 0)
+      {
+        success = false;
+      }
+    }
+    else
+    {
+      success = false;
+    }
+
+    for (size_t i = 0; i < posix_signals.size(); ++i)
+    {
+      struct sigaction action;
+      memset(&action, 0, sizeof action);
+      action.sa_flags = static_cast<int>(SA_SIGINFO | SA_ONSTACK | SA_NODEFER | SA_RESETHAND);
+      sigfillset(&action.sa_mask);
+      sigdelset(&action.sa_mask, posix_signals[i]);
+#if defined(__clang__)
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdisabled-macro-expansion"
+#endif
+      action.sa_sigaction = &sig_handler;
+#if defined(__clang__)
+#pragma clang diagnostic pop
+#endif
+
+      int r = sigaction(posix_signals[i], &action, nullptr);
+      if (r < 0)
+        success = false;
+    }
+
+    _loaded = success;
+  }
+
+  bool loaded() const
+  {
+    return _loaded;
+  }
+
+  static void handleSignal(int, siginfo_t *info, void *_ctx)
+  {
+    ucontext_t *uctx = static_cast<ucontext_t *>(_ctx);
+
+    backward::StackTrace st;
+    void *               error_addr = nullptr;
+#ifdef REG_RIP  // x86_64
+    error_addr = reinterpret_cast<void *>(uctx->uc_mcontext.gregs[REG_RIP]);
+#elif defined(REG_EIP)  // x86_32
+    error_addr = reinterpret_cast<void *>(uctx->uc_mcontext.gregs[REG_EIP]);
+#elif defined(__arm__)
+    error_addr = reinterpret_cast<void *>(uctx->uc_mcontext.arm_pc);
+#elif defined(__aarch64__)
+    error_addr = reinterpret_cast<void *>(uctx->uc_mcontext.pc);
+#elif defined(__ppc__) || defined(__powerpc) || defined(__powerpc__) || defined(__POWERPC__)
+    error_addr = reinterpret_cast<void *>(uctx->uc_mcontext.regs->nip);
+#elif defined(__s390x__)
+    error_addr = reinterpret_cast<void *>(uctx->uc_mcontext.psw.addr);
+#elif defined(__APPLE__) && defined(__x86_64__)
+    error_addr = reinterpret_cast<void *>(uctx->uc_mcontext->__ss.__rip);
+#elif defined(__APPLE__)
+    error_addr = reinterpret_cast<void *>(uctx->uc_mcontext->__ss.__eip);
+#else
+#warning ":/ sorry, ain't know no nothing none not of your architecture!"
+#endif
+    if (error_addr)
+    {
+      st.load_from(error_addr, 32);
+    }
+    else
+    {
+      st.load_here(32);
+    }
+
+    backward::Printer printer;
+    printer.address = true;
+
+    std::ostringstream oss;
+
+    printer.print(st, oss);
+
+    FETCH_LOG_ERROR("SIG_HANDLER", oss.str());
```

Comment:
Basically the same functionality as the normal sig handler from backward-cpp, except using the fetch logger instead of cerr